src/runner/index.js

```diff
@@ -24,13 +24,15 @@ export default class Runner extends EventEmitter {
         };
     }
 
-    // Run task
-    async _runTask (reporterPlugin, browserSet, tests) {
-        var task     = new Task(tests, browserSet.connections, this.proxy, this.opts);
-        var reporter = new Reporter(reporterPlugin, task, this.opts.reportOutStream);
+    static async _disposeTaskAndBrowsers (task, browserSet) {
+        task.abort();
+        task.removeAllListeners();
 
-        task.on('browser-job-done', job => browserSet.releaseConnection(job.browserConnection));
+        await browserSet.dispose();
+    }
 
+    // Run task
+    async _waitTaskResult (task, browserSet, reporter) {
```

Comment:
Maybe `getTaskResult` since it doesn't waits actually, but returns promise for the task result